packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -324,6 +337,56 @@ export const useInvokeAPI: EnsembleActionHook<InvokeAPIAction> = (action) => {
 
     const fireRequest = async (): Promise<void> => {
       setIsLoading(true);
+      // First check if useMockResponse is enabled and if a mockResponse exists
+      if (api.mockResponse && useMockResponse) {
+        // Ensure that the mock response contains a correctly formatted
+        if (typeof evaluatedMockResponse.response !== "object")
+          throw new Error(
+            "Improperly formatted mock response: Malformed mockResponse object",
+          );
+
+        if (
+          !evaluatedMockResponse.response.statusCode ||
+          typeof evaluatedMockResponse.response.statusCode !== "number"
+        )
+          throw new Error(
+            "Improperly formatted mock response: Incorrect Status Code. Please check that you have included a status code and that it is a number",
+          );
+
+        const isSuccess: boolean =
+          evaluatedMockResponse.response.statusCode >= 200 &&
+          evaluatedMockResponse.response.statusCode <= 299;
+        const mockResponse = {
+          ...evaluatedMockResponse.response,
+          isLoading: false,
+          isSuccess,
+          isError: !isSuccess,
+          appContext,
+        };
+
+        setData(api.name, mockResponse);
+
+        if (action?.id) {
+          setData(action.id, mockResponse);
+        }
+
+        if (isSuccess) {
+          onAPIResponseAction?.callback({ ...context, response: mockResponse });
+          onInvokeAPIResponseAction?.callback({
+            ...context,
+            response: mockResponse,
+          });
+        } else {
+          onAPIErrorAction?.callback({ ...context, error: mockResponse });
+          onInvokeAPIErrorAction?.callback({ ...context, error: mockResponse });
+        }
+
+        setIsComplete(true);
+        setIsLoading(false);
```

Comment:
Let's put all the mock handling logic into a separate function in a new `mock` module. You should be able to call the utility from the fetcher as well.